src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -112,6 +115,25 @@ public ResponseEntity<Page<AdvertisementTitleDto>> getPageOfAdvertisementsByKeyw
                 new ResponseEntity<>(allByKeyword, HttpStatus.OK);
     }
 
+    @GetMapping(ApiKey.ADV_SEARCH_PAGINATED_BY_CATEGORY_ID)
+    @Operation(summary = "Find advertisements by category id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
+            @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+    public Page<AdvertisementTitleDto> getPageOfAdvertisementsByCategoryId(
```

Comment:
You don't have to use **page** in the method's name because the return type describes it better.
It'd be better to rename the method to **findAdvertisementsHavingCategory**

Reply:
rename